e2e/angular-core/src/module-federation.test.ts

```diff
@@ -0,0 +1,279 @@
+    const port1 = 4200;
+    const port2 = 4206;
+    const hostApp = uniq('app');
+    const remoteApp1 = uniq('remote');
+    const sharedLib = uniq('shared-lib');
+    const secondaryEntry = uniq('secondary');
+
+    // generate host app
+    runCLI(
+      `generate @nrwl/angular:host ${hostApp} --style=css --no-interactive`
+    );
+
+    // generate remote apps
+    runCLI(
+      `generate @nrwl/angular:remote ${remoteApp1} --host=${hostApp} --port=${port2} --style=css --no-interactive`
+    );
+
+    // generate a shared lib
+    runCLI(
+      `generate @nrwl/angular:library ${sharedLib} --buildable --no-interactive`
+    );
+    runCLI(
+      `generate @nrwl/angular:library-secondary-entry-point --library=${sharedLib} --name=${secondaryEntry} --no-interactive`
+    );
+
+    // update the files to use shared library
+    updateFile(
+      `apps/${hostApp}/src/app/app.module.ts`,
+      `import { NgModule } from '@angular/core';
+      import { BrowserModule } from '@angular/platform-browser';
+      import { ${
+        names(sharedLib).className
+      }Module } from '@${proj}/${sharedLib}';
+      import { ${
+        names(secondaryEntry).className
+      }Module } from '@${proj}/${secondaryEntry}';
+      import { AppComponent } from './app.component';
+      import { NxWelcomeComponent } from './nx-welcome.component';
+      import { RouterModule } from '@angular/router';
+      
+      @NgModule({
+        declarations: [AppComponent, NxWelcomeComponent],
+        imports: [
+          BrowserModule,
+          SharedModule,
+          RouterModule.forRoot(
+            [
+              {
+                path: '${remoteApp1}',
+                loadChildren: () =>
+                  import('${remoteApp1}/Module').then(
+                    (m) => m.RemoteEntryModule
+                  ),
+              },
+            ],
+            { initialNavigation: 'enabledBlocking' }
+          ),
+        ],
+        providers: [],
+        bootstrap: [AppComponent],
+      })
+      export class AppModule {}
+      `
+    );
+    updateFile(
+      `apps/${remoteApp1}/src/app/remote-entry/entry.module.ts`,
+      `import { NgModule } from '@angular/core';
+    import { CommonModule } from '@angular/common';
+    import { RouterModule } from '@angular/router';
+    import { ${names(sharedLib).className}Module } from '@${proj}/${sharedLib}';
+      import { ${
+        names(secondaryEntry).className
+      }Module } from '@${proj}/${secondaryEntry}';
+    import { RemoteEntryComponent } from './entry.component';
+    
+    @NgModule({
+      declarations: [RemoteEntryComponent],
+      imports: [
+        CommonModule,
+        SharedModule,
+        RouterModule.forChild([
+          {
+            path: '',
+            component: RemoteEntryComponent,
+          },
+        ]),
+      ],
+      providers: [],
+    })
+    export class RemoteEntryModule {}
+    `
+    );
+
+    let process: ChildProcess;
+
+    try {
```

Comment:
I think we can just do `const process = await runCommandUntil(...)` since any errors thrown should cause the test to fail anyway.